### PR TITLE
Running Without Activity

### PR DESCRIPTION
**Pull request by timhuff**
I made this modification in order to solve issue #3

### CODE REVIEW

**Comment by timhuff** [comment]
I also modified the README.md, if you'd like to see what the API looks like. I modified `EasyLocationDelegate` to use this API.

**Comment by akhgupta** [comment]
Thanks Tim, Seems like big Pull request, I will review it.

**Comment by timhuff** [comment]
Thanks, @akhgupta. It is a bit of a large modification - I got a bit carried away with the API design. I think it came out pretty well, though. Let me know if you feel it needs to be modified.

**Comment by timhuff** [comment]
It might be good to document in the README the fact that calling `.get*Client(...)` a 2nd time will cause the 1st client to stop. I felt this was a necessary precaution due to the fact that only one background service will run at a time.

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+
+public class EasyLocationClientFactory {
+    private Context context;
+    private static EasyLocationClient currentEasyLocationClient;
```
Comment (akhgupta) [inline android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java:15]: @timhuff static fields can create memory leaks. Android lint is giving warnings for these too

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java [hunk 2]
```diff
@@ -210,30 +143,50 @@ void onRequestPermissionsResult(int requestCode, int[] grantResults) {
...
+                    @Override
+                    public void onLocationReceived(Location location) {
+                        easyLocationListener.onLocationReceived(location);
+                    }
+                });
+
     }
 
     void requestSingleLocationFix(EasyLocationRequest easyLocationRequest) {
-        isProperRequest(easyLocationRequest);
```
Comment (akhgupta) [inline android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java:236]: @timhuff isProperRequest is required to help developers integrating this library.

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java (hunk 1 above)
Comment (timhuff) [reply android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java:15]: Wait... static fields in general? How are you supposed to make a singleton without static fields?

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java (hunk 2 above)
Comment (timhuff) [reply android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java:236]: My mistake. I think I meant to move that to the new component. I'll fix that up.

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java [hunk 3]
```diff
@@ -147,59 +117,22 @@ private void requestPermission() {
...
-    void onCreate() {
-        mLocationManager = (LocationManager) activity.getSystemService(Context.LOCATION_SERVICE);
-        googleApiAvailability = GoogleApiAvailability.getInstance();
-        registerLocationBroadcastReceiver();
-    }
-
     void onActivityResult(int requestCode) {
         switch (requestCode) {
             case ENABLE_LOCATION_SERVICES_REQUEST:
-                if (isLocationEnabled()) {
```
Comment (akhgupta) [inline android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java:200]: @timhuff isLocationEnabled already does this stuff, looks prettier

File: android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java (hunk 1 above)
Comment (akhgupta) [reply android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationClientFactory.java:15]: No, but this class has Context as field, which can cause Activity memory leak.

**Comment by Savrov** [comment]
Hi, is this request approved?

**Comment by akhgupta** [comment]
Unfortunately no, it has some unresolved discussion points.

**Pull request closed without merging**